### PR TITLE
Add a rich text input for autocomponents and apply nice polaris styling

### PR DESCRIPTION
**Pull request by airhorns**
- uses https://mdxeditor.dev/editor/docs/getting-started which seems like a nice, medium weight markdown editor with lightweight customizability
 - lazyloads this component using `React.lazy` for smaller default bundle size for autocomponents, as richtext is not often used that much
 - wires up mdxeditor's base toolbar to show nice formatting buttons
 - styles the rendered markdown using polaris css variables so that it matches kinda nicely in polaris

[no-changelog-required]

### CODE REVIEW

**Review by danroberts — APPROVED** [review]
👍 on 🟢

File: packages/react/src/auto/shared/AutoRichTextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  props: {
+    field: string;
+    control?: Control<any>;
+    editorRef?: ForwardedRef<MDXEditorMethods> | null;
+  } & Omit<MDXEditorProps, "markdown" | "onChange">
+) {
+  const { formState } = useFormContext();
+  const { field, control, editorRef, ...rest } = props;
+  const controller = useStringInputController({ field, control });
+
```
Comment (MillanWangGadget) [inline packages/react/src/auto/shared/AutoRichTextInput.tsx:39]: - `useFormContext` is not needed when you have the controller
  - `controller.value?.markdown === get(formState.defaultValues, field)?.markdown`
- Add this before returning MDXEditor
  - `if (!controller.value?.markdown) {   return null;  }`
  - Fixes the `loading in without the existing record value` issue

File: packages/react/src/auto/shared/AutoRichTextInput.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    <MDXEditor
+      plugins={[
+        headingsPlugin(),
+        listsPlugin(),
+        quotePlugin(),
+        thematicBreakPlugin(),
+        markdownShortcutPlugin(),
+        linkDialogPlugin(),
+        diffSourcePlugin({
+          diffMarkdown: get(formState.defaultValues, field)?.markdown ?? "",
```
Comment (MillanWangGadget) [inline packages/react/src/auto/shared/AutoRichTextInput.tsx:50]: Can be replaced with `controller.value?.markdown`

File: packages/react/src/auto/shared/AutoRichTextInput.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+              <BoldItalicUnderlineToggles />
+              <ListsToggle />
+              <CodeToggle />
+              <CreateLink />
+            </DiffSourceToggleWrapper>
+          ),
+        }),
+      ]}
+      contentEditableClassName="autoform-prose"
+      {...rest}
```
Comment (MillanWangGadget) [inline packages/react/src/auto/shared/AutoRichTextInput.tsx:69]: Perhaps this should be moved to the absolute bottom in the event that the user wants to override `markdown, onChange, ref`

**Review by MillanWangGadget — COMMENTED** [review]
- 🍏 In the dependencies deep end, I see.... Things look reasonable 👍 
- 🟡 Loading existing record values is race-y. Sometimes the input is not populated with the record value
  - CleanShot 2024-07-16 at 16 20 03
  - See code comment for easy fix 😎 

Some code comments but good otherwise 👍

**Comment by MillanWangGadget** [comment]
Loading existing record values is reliable now 👍

**Pull request merged**